app/views/questionnaire.py

```diff
@@ -40,21 +41,26 @@
                                     import_name=__name__,
                                     url_prefix='/questionnaire/<eq_id>/<form_type>/<collection_id>/')
 
+after_submission_blueprint = Blueprint(name='after_submission',
+                                       import_name=__name__,
+                                       url_prefix='/questionnaire')
+
 
 @questionnaire_blueprint.before_request
 def before_request():
     metadata = get_metadata(current_user)
-    if metadata:
-        logger.bind(tx_id=metadata['tx_id'])
+    if metadata is None:
+        raise NoTokenException(401)
```

Comment:
I do wonder if we are trying to re-implement authentication logic.

Reply:
We are really as at this point the user is still authenticated but we don't want to allow them back into the questionnaire. It may be worth creating a new error for this.

Reply:
Do we have the submitted flag/time in the session? if so can we check this and redirect them to the thankyou?

Reply:
This matches current behaviour if a user attempts to get back into the questionnaire after submission. Another one to discuss with @ajmaddaford. Argument for both but confusion could arise if the user is pressing back but just keeps getting routed to where they were in the first place without any kind of notification.